dandi/metadata.py

```diff
@@ -59,64 +68,90 @@ def get_metadata(path: Union[str, Path]) -> Optional[dict]:
     -------
     dict
     """
+
+    from .files import bids, dandi_file, find_bids_dataset_description
+
     # when we run in parallel, these annoying warnings appear
     ignore_benign_pynwb_warnings()
-    path = str(path)  # for Path
+    path = os.path.abspath(str(path))  # for Path
     meta = dict()
 
-    if op.isdir(path):
+    if op.isdir(path) and not path.endswith(tuple(ZARR_EXTENSIONS)):
         try:
             dandiset = Dandiset(path)
             return cast(dict, dandiset.metadata)
         except ValueError as exc:
             lgr.debug("Failed to get metadata for %s: %s", path, exc)
             return None
 
-    if nwb_has_external_links(path):
-        raise NotImplementedError(
-            f"NWB files with external links are not supported: {path}"
+    # Is the data BIDS (as defined by the presence of a BIDS dataset descriptor)
+    bids_dataset_description = find_bids_dataset_description(path)
+    if bids_dataset_description:
+        dandiset_path = find_parent_directory_containing("dandiset.yaml", path)
+        bids_dataset_description = find_bids_dataset_description(path)
+        df = dandi_file(
+            Path(path),
+            dandiset_path,
+            bids_dataset_description=bids_dataset_description,
         )
+        path_metadata = df.get_metadata(digest=digest)
+        assert isinstance(df, bids.BIDSAsset)
+        meta["bids_version"] = df.get_validation_bids_version()
+        # there might be a more elegant way to do this:
+        for key in metadata_all_fields:
+            try:
+                value = getattr(path_metadata.wasAttributedTo[0], key)
+            except AttributeError:
+                pass
+            else:
+                meta[key] = value
+    elif path.endswith((".NWB", ".nwb")):
```

Comment:
```suggestion
    if path.endswith((".NWB", ".nwb")):
```

in relation to prior comment https://github.com/dandi/dandi-cli/pull/1164#discussion_r1042649545  -- the point is to validate .nwb regardless either they are part of BIDS or not.  BIDS validation does not do any validation of .nwb files which could be within BIDS dataset.